server/src/main/java/org/opensearch/common/blobstore/BlobContainer.java

```diff
@@ -80,17 +80,17 @@ public interface BlobContainer {
     InputStream readBlob(String blobName) throws IOException;
 
     /**
-     * Creates a new {@link FetchBlobResult} for the given blob name.
+     * Creates a new {@link InputStreamWithMetadata} for the given blob name. The default implementation is null; the implementing class should provide its own implementation.
      *
      * @param   blobName
-     *          The name of the blob to get an {@link InputStream} for.
-     * @return  The {@link FetchBlobResult} of the blob.
+     *          The name of the blob to get an {@link InputStreamWithMetadata} for.
+     * @return  The {@link InputStreamWithMetadata} of the blob.
      * @throws  NoSuchFileException if the blob does not exist
      * @throws  IOException if the blob can not be read.
      */
     @ExperimentalApi
-    default FetchBlobResult readBlobWithMetadata(String blobName) throws IOException {
-        throw new UnsupportedOperationException("readBlobWithMetadata is not implemented yet");
+    default InputStreamWithMetadata readBlobWithMetadata(String blobName) throws IOException {
+        return null;
```

Comment:
Why can't we just throw UnsupportedOperationException as the default behaviour?

Reply:
right. Made changes.